File: IntroduccionPOO/Ejercicio1/Persona.java

package introduccionpoo;

public class Persona {
    private String nombre;
    private int edad;
    private String ciudad;

    
    public Persona(String nombre, int edad, String ciudad) {
        this.nombre = nombre;
        this.edad = edad;
        this.ciudad = ciudad;
    }

    public void saludar() {
        System.out.println("Hola, soy " + nombre + " de " + ciudad);
    }

    public boolean esMayorDeEdad() {
        return edad >= 18;
    }

    
    public static void main(String[] args) {
      
        Persona persona1 = new Persona("Noemí", 20, "  rio seco");
        Persona persona2 = new Persona("Chino", 17, "peru");
        Persona persona3 = new Persona("Anahi", 25, "muñecas");

       
        persona1.saludar();
        persona2.saludar();
        persona3.saludar();

        System.out.println(persona1.nombre + " es mayor de edad: " + persona1.esMayorDeEdad());
        System.out.println(persona2.nombre + " es mayor de edad: " + persona2.esMayorDeEdad());
        System.out.println(persona3.nombre + " es mayor de edad: " + persona3.esMayorDeEdad());
        System.out.println("noemi chino blanco");
    }
}

File: IntroduccionPOO/Ejercicio1/persona.py
class Persona:
    def __init__(self, nombre, edad, ciudad):
        self.nombre = nombre
        self.edad = edad
        self.ciudad = ciudad

    def saludar(self):
        print(f"Hola, soy {self.nombre} de {self.ciudad}")

    def es_mayor_de_edad(self):
        return self.edad >= 18

    def mostrar_mayor_de_edad(self):
        if self.es_mayor_de_edad():
            print(f"{self.nombre} es mayor de edad.")
        else:
            print(f"{self.nombre} NO es mayor de edad.")


p1 = Persona("Noemi Chino blanco", 20, "La Paz")
p2 = Persona("Jose", 19, "Tarija")
p3 = Persona("Alejandro", 22, "Pando")

p1.saludar()
p1.mostrar_mayor_de_edad()
print()

p2.saludar()
p2.mostrar_mayor_de_edad()
print()

p3.saludar()
p3.mostrar_mayor_de_edad()
